packages/react/src/Footnotes/Footnotes.tsx

```diff
@@ -0,0 +1,112 @@
+import React from 'react'
+import {ReplyIcon} from '@primer/octicons-react'
+import clsx from 'clsx'
+
+import {Heading, InlineLink, Text} from '../'
+import type {BaseProps} from '../component-helpers'
+
+/**
+ * Design tokens
+ */
+import '@primer/brand-primitives/lib/design-tokens/css/tokens/functional/components/footnotes/colors-with-modes.css'
+
+/** * Main Stylesheet (as a CSS Module) */
+import styles from './Footnotes.module.css'
+
+export const FootnotesTags = ['div', 'ol'] as const
+
+type FootnotesRestrictedPolymorphism =
+  | (React.HTMLAttributes<HTMLDivElement> & BaseProps<HTMLDivElement> & {as?: 'div'})
+  | (React.HTMLAttributes<HTMLParagraphElement> & BaseProps<HTMLOListElement> & {as?: 'ol'})
+
```

Comment:
I don't think we need `BaseProps` here do we? I don't think it's accurate either as we don't support the `animate` prop.

Reply:
BaseProps is a general-purpose wrapper around all components in Primer Brand, so I'd prefer to leave that in unless it's causing problems. It gives us the options to centralise certain props in future. Fair point around `animate`. The context there is that _every_ Primer Brand component was meant to animate in the past, but at some point we decided that shouldn't be the case. I think we should consider deprecating it in future. Will add to maintainer sync agenda.

Reply:
Interesting, thanks. I had wondered about `BaseProps` in the past because it doesn't really seem to do anything other than add the `animate` prop. Assuming all of our components extend `HTMLAttributes<HTMLWhateverElement>` then they already have `id`, `className`, and `ref` types.